pkg/controller/nexus/resource/deployment/manager.go

```diff
@@ -40,11 +41,11 @@ type Manager struct {
 	client client.Client
 }
 
-// NewManager creates a deployment resources manager.
+// NewManager creates a deployment resources manager
 // It is expected that the Nexus has been previously validated.
-func NewManager(nexus v1alpha1.Nexus, client client.Client) *Manager {
+func NewManager(nexus *v1alpha1.Nexus, client client.Client) *Manager {
```

Comment:
Actually that is a bug since you were changing the `spec.*` attributes. Those are not getting updated in the main reconcile loop when setting defaults. See the `networking` package, when on OCP we change the `ExposeAs` attribute to `Route`. That was not being updated since we were changing the copy of the object, not the reference.

Reply:
Oh man, ok, even nicer catch